test/e2e/tests/vision/vision.spec.ts

```diff
@@ -0,0 +1,144 @@
+import {expect} from '@playwright/test'
+import {test} from '@sanity/test'
+
+import {encodeQueryString, getVisionRegions, openVisionTool} from './utils'
+
+test.describe('Vision', () => {
+  test('should be possible to type an execute a query', async ({page, sanityClient}) => {
+    const bookTitle = 'Test Book'
+    const bookDocument = await sanityClient.create({
+      _type: 'book',
+      title: bookTitle,
+    })
+
+    await openVisionTool(page)
+    // Clears local storage
+    await page.evaluate(() => localStorage.clear())
+
+    const {queryEditor, paramsEditor, paramsRegion, resultRegion} = await getVisionRegions(page)
+
+    // Click to focus the editor
+    await queryEditor.click()
+
+    // Type text into the CodeMirror editor
+    const inputText = '*[_type == "book" && _id == $id]{_id, title}'
+    await queryEditor.fill(inputText)
+
+    // Assert that the text was correctly inserted
+    await expect(queryEditor).toHaveText(inputText)
+
+    const paramsInputText = JSON.stringify({id: bookDocument._id})
+    // Type text into the params editor
+    await paramsEditor.fill(paramsInputText.slice(0, -2))
+    // Error icon should be visible
+    await expect(paramsRegion.locator('[data-sanity-icon="error-outline"]')).toBeVisible()
+
+    // Fill the params editor with the correct text
+    await paramsEditor.fill(paramsInputText)
+    // Error icon should not be visible
+    await expect(paramsRegion.locator('[data-sanity-icon="error-outline"]')).not.toBeVisible()
+    // Assert that the text was correctly inserted
+    await expect(paramsEditor).toHaveText(paramsInputText)
+
+    // Find the button with the text "Fetch" and click it.
+    await page.locator('button').filter({hasText: 'Fetch'}).click()
+
+    // Assert that the results are visible
+    // It should find the book document assert that by checking the title and the id
+    await expect(resultRegion.getByText(bookTitle)).toBeVisible()
+    await expect(resultRegion.getByText(bookDocument._id)).toBeVisible()
```

Comment:
Might be just a preference, but I would usually prefer that we match to a snapshot / to the text expected than whether something is visible

Reply:
After clarificiation, this is not needed for this particular test!